compiler/src/iree/compiler/Codegen/Common/FastMathDeviceLib.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/FastMathDeviceLib.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/IR/PatternMatch.h"
+
+namespace mlir::iree_compiler {
+
+namespace {
+// Pattern to lower math.erf to its device lib implementation (from erfF.cl)
+struct ErfDeviceLibPattern : public OpRewritePattern<math::ErfOp> {
```

Comment:
these aren't library calls, they are just patterns that implement an op, so you can drop the device/lib - given that we actually have use of device libraries it's confusing here - I was about to route this PR out of the Common/ dir because something using a device library is target-specific, but this isn't (which is great, hooray reusable code!)

```suggestion
struct FastErfPattern : public OpRewritePattern<math::ErfOp> {
```

Reply:
@benvanik Thanks for the review! I've addressed your comments.